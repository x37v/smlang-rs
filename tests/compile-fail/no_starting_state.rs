extern crate smlang;

use smlang::statemachine;

#[derive(PartialEq)]
pub enum Events {
    Event1,
    Event2,
}

pub struct Context;

statemachine! {
    transitions: {
        //~ ERROR No starting state defined, indicate the starting state with a *
        State1 + Event1 = State2,
        State2 + Event2 = State3,
    }
}

fn main() {}
