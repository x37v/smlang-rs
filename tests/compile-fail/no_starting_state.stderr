error: No starting state defined, indicate the starting state with a *.
  --> tests/compile-fail/no_starting_state.rs:13:1
   |
13 | / statemachine! {
14 | |     transitions: {
15 | |         //~ ERROR No starting state defined, indicate the starting state with a *
16 | |         State1 + Event1 = State2,
17 | |         State2 + Event2 = State3,
18 | |     }
19 | | }
   | |_^
   |
   = note: this error originates in the macro `statemachine` (in Nightly builds, run with -Z macro-backtrace for more info)
